src/mainwindow/mainwindow.cpp

```diff
@@ -427,18 +601,92 @@ void MainWindow::onCheckUpdatesTriggered()
 
 void MainWindow::onOpenFileFromProjectViewer(QString fileName)
 {
+    // if document already opened then return
+    if(checkIfOpened(fileName))
+    {
+        QMessageBox::warning(this, "Document already opened", "Selected document already opened.");
+        return;
+    }
+
     openDoc(fileName);
 }
 
+void MainWindow::onCloseWindow(CodeEditor *curDoc)
+{
+    if(!curDoc)
+        return;
+
+    if(curDoc->document()->isModified())
+    {
+        QMessageBox::StandardButton reply = QMessageBox::question
+                (
+                    this,
+                    "Saving Changes",
+                    "Do you want to save changes to opened documents?",
+                    QMessageBox::Yes | QMessageBox::No
+                    );
+
+        if(reply == QMessageBox::No)
+            return;
+
+        saveDocument(curDoc, curDoc->getFileName());
+    }
+}
+
 CodeEditor* MainWindow::createNewDoc()
```

Comment:
Do you use this function?

Reply:
Every time we open file / create new file new instance of CodeEditor is created via method createNewDoc
Should I refactor this?